api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala

```diff
@@ -144,39 +144,52 @@ trait InstitutionsHttpApi extends InstitutionProtocol with ApiErrorProtocol with
-                  "uploaded"
-                }
-              case Failure(error) =>
-                processingActor ! Shutdown
-                log.error(error.getLocalizedMessage)
-                val errorResponse = ErrorResponse(400, "Invalid File Format", path)
-                complete(ToResponseMarshallable(StatusCodes.BadRequest -> errorResponse))
-            }
-          }
+      time {
+        val path = s"institutions/$institutionId/filings/$period/submissions/$submissionId"
+        extractExecutionContext { executor =>
+          val uploadTimestamp = Instant.now.toEpochMilli
+          val processingActor = createHmdaRawFile(system, submissionId)
+          val submissionsActor = system.actorOf(SubmissionPersistence.props(institutionId, period))
+          implicit val ec: ExecutionContext = executor
+          val fIsSubmissionOverwrite = checkSubmissionOverwrite(submissionsActor, submissionId.toInt)
+          onComplete(fIsSubmissionOverwrite) {
+            case Success(false) =>
+              submissionsActor ! Shutdown
+              processingActor ! StartUpload
+              fileUpload("file") {
+                case (metadata, byteSource) if (metadata.fileName.endsWith(".txt")) =>
+                  val uploadedF = byteSource
+                    .via(splitLines)
+                    .map(_.utf8String)
+                    .runForeach(line => processingActor ! AddLine(uploadTimestamp, line))
 
-        case _ =>
-          time {
-            processingActor ! Shutdown
-            val errorResponse = ErrorResponse(400, "Invalid File Format", path)
-            complete(ToResponseMarshallable(StatusCodes.BadRequest -> errorResponse))
+                  onComplete(uploadedF) {
+                    case Success(response) =>
+                      processingActor ! CompleteUpload
+                      processingActor ! Shutdown
+                      complete {
+                        "uploaded"
+                      }
+                    case Failure(error) =>
+                      processingActor ! Shutdown
+                      log.error(error.getLocalizedMessage)
+                      val errorResponse = ErrorResponse(400, "Invalid File Format", path)
+                      complete(ToResponseMarshallable(StatusCodes.BadRequest -> errorResponse))
+                  }
+                case _ =>
+                  processingActor ! Shutdown
+                  val errorResponse = ErrorResponse(400, "Invalid File Format", path)
+                  complete(ToResponseMarshallable(StatusCodes.BadRequest -> errorResponse))
+              }
+            case Success(true) =>
+              val errorResponse = ErrorResponse(400, "Submission already exists", path)
+              complete(ToResponseMarshallable(StatusCodes.BadRequest -> errorResponse))
+            case Failure(_) =>
+              submissionsActor ! Shutdown
+              val errorResponse = ErrorResponse(500, "Internal server error", path)
```

Comment:
Now that #493 merged, you can use `completeWithInternalError` for this 500 error